backend/owa.ts

```diff
@@ -88,6 +92,27 @@ export async function fetchJSON(partition: string, url: string, action: string,
   return result;
 }
 
+export async function streamJSON(partition: string, url: string) {
+  let result = {
+    ok: false,
+    status: 0,
+    statusText: '',
+    body: null,
+  };
+  let session = Session.fromPartition(partition);
+  let cookies = await session.cookies.get({ name: 'X-OWA-CANARY' });
```

Comment:
It seems to be that this cookie getter could be moved into a separate function and the canary added to the URL by the caller.

We want to keep the backend functions absolutely minimal and generic. All protocol implementation details (like canary) should be in app/logic/, not in backend.ts.

By refactoring it this way, you may be able to unite functions `streamJSON()` and `streamHTTP()`.

Reply:
The caller is JPC.

Reply:
I understand your reason is that you want to avoid a JPC-round-trip to get the cookies via JPC, then calling the HTTP function via JPC as well, for every OWA server call.